src/NServiceBus.Core/Hosting/Helpers/AssemblyScanner.cs

```diff
@@ -54,31 +54,46 @@ internal AssemblyScanner(Assembly assemblyToScan)
         public AssemblyScannerResults GetScannableAssemblies()
         {
             var results = new AssemblyScannerResults();
-
             var processed = new Dictionary<string, bool>(StringComparer.OrdinalIgnoreCase);
+
             if (assemblyToScan != null)
             {
-                ScanAssemblyAndDependencies(assemblyToScan, processed, results);
+                if (ScanAssembly(assemblyToScan, processed))
+                {
+                    AddTypesToResult(assemblyToScan, results);
```

Comment:
so in this case we would no longer include types from a referenced assembly of the target assembly, but I guess that's ok. The only usage only passes core and wants to receive core's types, so there shouldn't be an issue